include/ua_securitycontext.h

```diff
@@ -0,0 +1,196 @@
+/*
+* Copyright (C) 2014 the contributors as stated in the AUTHORS file
+*
+* This file is part of open62541. open62541 is free software: you can
+* redistribute it and/or modify it under the terms of the GNU Lesser General
+* Public License, version 3 (as published by the Free Software Foundation) with
+* a static linking exception as stated in the LICENSE file provided with
+* open62541.
+*
+* open62541 is distributed in the hope that it will be useful, but WITHOUT ANY
+* WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+* A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+* details.
+*/
+
+#ifndef UA_SECURITYCONTEXT_H_
+#define UA_SECURITYCONTEXT_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "ua_types.h"
+#include "ua_log.h"
+#include "ua_securitypolicy_fwd.h"
+#include "ua_securitycontext_fwd.h"
+
+struct UA_Endpoint_SecurityContext
+{
+    UA_StatusCode (*const init)(const UA_SecurityPolicy *const securityPolicy,
+                                const void *const initData,
+                                void **const pp_contextData);
+
+    UA_StatusCode (*const deleteMembers)(const UA_SecurityPolicy *const securityPolicy,

```

Comment:
We could reduce the complexity here a little.
`const UA_SecurityPolicy *const securityPolicy` makes users think hard about C type declarations and what is const when.
https://cdecl.org/?q=const+struct+UA_SecurityPolicy+*const+securityPolicy

`const UA_SecurityPolicy *securityPolicy` (without the second const) should be enough.
After all, this is an interface definition. And the caller does not care if securityPolicy points to a different struct at some time, when it's purely internal to the pointed method.